ports/raspberrypi/common-hal/rp2pio/StateMachine.c

```diff
@@ -228,15 +228,19 @@ bool rp2pio_statemachine_construct(rp2pio_statemachine_obj_t *self,
     sm_config_set_clkdiv_int_frac(&c, div256 / 256, div256 % 256);
 
     if (first_out_pin != NULL) {
+        pio_sm_set_consecutive_pindirs(self->pio, self->state_machine, first_out_pin->number, out_pin_count, true);
         sm_config_set_out_pins(&c, first_out_pin->number, out_pin_count);
     }
     if (first_in_pin != NULL) {
+        pio_sm_set_consecutive_pindirs(self->pio, self->state_machine, first_in_pin->number, in_pin_count, false);
         sm_config_set_in_pins(&c, first_in_pin->number);
     }
     if (first_set_pin != NULL) {
+        pio_sm_set_consecutive_pindirs(self->pio, self->state_machine, first_set_pin->number, set_pin_count, true);
         sm_config_set_set_pins(&c, first_set_pin->number, set_pin_count);
     }
     if (first_sideset_pin != NULL) {
+        pio_sm_set_consecutive_pindirs(self->pio, self->state_machine, first_sideset_pin->number, sideset_pin_count, true);
```

Comment:
My plan was to require an init program to do this. I'm a little worried about enabling an input to an arbitrary level. Instead, you can provide the OUT pins as set and use set in init to set output level and output direction. I thought sideset was always output too.

Reply:
I ran into an issue trying to remove these. All the samples are `set pindirs 1` or `out pindirs 1` and from what I saw in the RP2040 docs it seems only 5 pins max set at once. Even when I run `out` with 0x1F and then 0x07 (for outputting 8 bits for 8 pins) it doesn't work. I tried to move my pin group to set as well as out groups and I cannot set 8 pin directions at once. The SDK code has the pins grouped into 5s and set that way. I'm thinking this may be a limit.

Every parallel pin example I can see seems to use to `pio_sm_set_consecutive_pindirs` command to set the pins. I'm at a loss now how else to set 8 `out` (or even `set`) pindirs.

I was able to successfully set the 1 `set` pin used for write in the `set` group.

Reply:
We talked about this in the meeting today and OUT should be able to set up the pin via the `init` sequence.